src/psmove.c

```diff
@@ -563,23 +563,47 @@ psmove_connect_internal(wchar_t *serial, char *path, int id)
     if (serial != NULL && wcslen(serial) > 1) {
         move->is_bluetooth = 1;
     }
-    /* Windows Quirk: Use path instead of serial number by ignoring serial */
-    serial = NULL;
+    serial = NULL;  // Set the serial to NULL, even if BT device, to use psmove_get_serial below.
 
-    /* XXX Ugly: Convert "col02" path to "col01" path (tested w/ BT and USB) */
+    /**
+     * In Windows, the device is enumerated 3 times, and each has slightly different behaviour.
+     * The devices' paths differ slightly (col01, col02, and col03).
+     * The device with col01 is the one we want to use for data.
+     * The device with col02 is the one we want to use for the bluetooth address.
+     * More testing remains to determine which device is best for which feature reports.
+     **/
+
+    /**
+     * We know this function (psmove_connect_internal) will only be called with the col01 path.
+     * We first copy that path then modify it to col02. That will be the path for our addr device.
+     * Connect to the addr device first, then connect to the main device.
+     **/
+    move->device_path_addr = strdup(path);
     char *p;
-    psmove_return_val_if_fail((p = strstr(path, "&col02#")) != NULL, NULL);
-    p[5] = '1';
-    psmove_return_val_if_fail((p = strstr(path, "&0001#")) != NULL, NULL);
-    p[4] = '0';
-#endif
+    psmove_return_val_if_fail((p = strstr(move->device_path_addr, "&col01#")) != NULL, NULL);
+    p[5] = '2';
+    psmove_return_val_if_fail((p = strstr(move->device_path_addr, "&0000#")) != NULL, NULL);
+    p[4] = '1';
+    move->handle_addr = hid_open_path(move->device_path_addr);
+    hid_set_nonblocking(move->handle_addr, 1);
+
+    move->device_path = strdup(path);
+    move->handle = hid_open_path(move->device_path);
 
+#else
+    /* If not in Windows then we can rely on having only one device. */
+    if (path != NULL) {
+        move->device_path = strdup(path);
+    }
     if (serial == NULL && path != NULL) {
         move->handle = hid_open_path(path);
-    } else {
+    }
+    else {
```

Comment:
Please be consistent in that file, and place the closing brace (`}`) in the same line as the `else`.